wallet/walletrpc.c

```diff
@@ -531,6 +531,58 @@ static const struct json_command listfunds_command = {
 };
 AUTODATA(json_command, &listfunds_command);
 
+static struct command_result *json_getbalance(struct command *cmd,
+					     const char *buffer,
+					     const jsmntok_t *obj UNNEEDED,
+					     const jsmntok_t *params)
+{
+    struct json_stream *response;
+    struct peer *p;
+    struct utxo **utxos;
+    struct amount_sat on_chain = AMOUNT_SAT(0), on_channels = AMOUNT_SAT(0);
+
+    if (!param(cmd, buffer, params, NULL))
+        return command_param_failed();
+
+    response = json_stream_success(cmd);
+    json_object_start(response, NULL);
+
+    /* Add funds from utxos */
+    utxos = wallet_get_utxos(cmd, cmd->ld->wallet, output_state_available);
+    for (size_t i = 0; i < tal_count(utxos); i++) {
+        if (utxos[i]->blockheight && !utxos[i]->spendheight) {
+            if (!amount_sat_add(&on_chain, on_chain, utxos[i]->amount))
+                return command_fail(cmd, LIGHTNINGD,
+                        "utxos addition failure.");
+        }
+    }
+    json_add_amount_sat(response, on_chain, "on_chain", "on_chain_msat");
+
+    /* Add funds that are allocated to channels */
+    list_for_each(&cmd->ld->peers, p, list) {
+        struct channel *c;
```

Comment:
you should move the declaration of `struct channel *c;` to the beginning of the function.